src/diffusion/ColoredGaussSeidelDiffuse.cpp

```diff
@@ -133,78 +113,69 @@ void ColoredGaussSeidelDiffuse::diffuse(
 /// \param  sync        synchronization boolean (true=sync (default), false=async)
 // ************************************************************************
 void ColoredGaussSeidelDiffuse::diffuse(
-        Field &out, const Field &, const Field &b,
+        Field &out, const Field &in, const Field &b,
         const real D, const Field &EV, bool sync) {
+    out.copy_data(in);  // cgs only calculates on out
     auto domain_data = DomainData::getInstance();
     auto domain_controller = DomainController::getInstance();
 
-    auto bsize_i = domain_controller->get_size_domain_inner_list_level_joined(0);
+    auto size_domain_inner_list = domain_controller->get_size_domain_inner_list_level_joined(0);
 
-    size_t* domain_inner_list = domain_controller->get_domain_inner_list_level_joined();
+    size_t *domain_inner_list = domain_controller->get_domain_inner_list_level_joined();
 
-//#pragma acc data present(d_out[:bsize], d_b[:bsize], d_EV[:bsize])
-{
-    const size_t Nx = domain_data->get_Nx();
-    const size_t Ny = domain_data->get_Ny();
+#pragma acc data present(out, b, EV)
+    {
+        const size_t Nx = domain_data->get_Nx();
+        const size_t Ny = domain_data->get_Ny();
```

Comment:
those variables aren't used in this function

Reply:
8fba1a3ca4c5c48d015a448dda7f852212fbe8da